x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java

```diff
@@ -349,6 +362,103 @@ public void testInvalidateApiKeys() throws Exception {
         assertThat(invalidateApiKeyResponse.getInvalidatedApiKeys(), emptyIterable());
     }
 
+    @SuppressWarnings("unchecked")
+    public void testInvalidateApiKeysWillFlipInvalidatedFlagAndRecordTimestamp() {
```

Comment:
This test is quite dense for what it does. I got a bit carried away in trying to have a unit level test. Happy to drop it if you prefer to having just the integration level test.

Reply:
I'm good with having this test -- it adds fast coverage and I don't think the maintenance burden is too high.